packages/adapter/src/snap.ts

```diff
@@ -25,11 +37,11 @@ export class MetamaskPolkadotSnap implements Injected {
   };
 
   public signer: InjectedSigner = {
-    signPayload: async (_payload) => {
-      return {id: 0, signature: ""};
+    signPayload: async (payload: SignerPayloadJSON): Promise<SignerResult> => {
+      return await signPayloadJSON(payload);
     },
-    signRaw: async (_raw) => {
-      return {id: 0, signature: ""};
+    signRaw: async (raw: SignerPayloadRaw): Promise<SignerResult> => {
+      return await signPayloadRaw(raw);
     },
     update: () => null
```

Comment:
Did you digure what's this method for?

Reply:
@mpetrunic I am investigating, inside definition it says:

`@description Receives an update for the extrinsic signed by a `signer.sign`

Do you have any idea what this id should represent? 

```
export interface SignerResult {
    /**
     * @description The id for this request
     */
    id: number;
    /**
     * @description The resulting signature in hex
     */
    signature: string;
}
```

Reply:
Not sure, cannot find anything in documentation, might be best to ask in riot chat.
My best guess is that you keep some counter on user signing requests so user can listen on updates in "update" method but not sure what kind up updates can there be